build_system/src/config.rs

```diff
@@ -273,6 +282,9 @@ impl ConfigInfo {
     }
 
     pub fn setup_gcc_path(&mut self) -> Result<(), String> {
+        if self.gcc_path.is_some() {
```

Comment:
Please add a comment to explain that this means the path was provided via `--gcc-path`, so the config file is not needed in this case.

Reply:
Would it make sense to also print a message saying "--gcc-path was provided, ignoring config.toml"?

Reply:
We don't display a message by default in this path, but I can display on in both cases to know where it comes from.

Reply:
Well, since this argument skips some code that usually run, I believe it would be useful to know that this is skipped.
I guess you can print a message in the other case, but I believe it's less useful since the behavior is less surprising.